src/java.base/share/classes/java/util/concurrent/ConcurrentHashMap.java

```diff
@@ -622,7 +626,7 @@
      * are special, and contain null keys and values (but are never
      * exported).  Otherwise, keys and vals are never null.
      */
-    static class Node<K,V> implements Map.Entry<K,V> {
+    static sealed class Node<K,V> implements Map.Entry<K,V> {
```

Comment:
There is a long term modernization task of making nested classes private to make use of
https://openjdk.org/jeps/181
It looks like this is an example.